omc3/utils/outliers.py

```diff
@@ -49,40 +84,40 @@ def get_filter_mask(data: ArrayLike, x_data: ArrayLike = None, limit: float = 0.
     if nsig is None:
         nsig = _get_significance_cut_from_length(np.sum(mask))
 
-    # To fulfill the condition for the first iteration:
-    prevlen = np.sum(mask) + 1
+    # Set number of remaining points to check decrease in loop
+    n_previous = np.sum(mask) + 1  # initialization for the first check
 
     # Cleaning iteration
     for _ in range(niter):
         # check that number of points decreases and some are left
-        if not ((np.sum(mask) < prevlen) and
-                (np.sum(mask) > 2)):
+        n_current = np.sum(mask)
+        if not ((n_current < n_previous) and (n_current > 2)):
             break
-        prevlen = np.sum(mask)
+        n_previous = n_current
 
         # get the (remaining) data-points used to get moments from,
         # if x-data is given, fit and remove slope.
         if x_data is None:
-            y, y_orig = _get_data(mask, data)
+            y, y_orig = data[mask], data
         else:
             y, y_orig = _get_data_without_slope(mask, x_data, data)
 
         # keep only points within nsig sigma of the remaining data.
         avg, std = np.mean(y), np.std(y)
         mask = np.logical_and(mask, np.abs(y_orig - avg) < np.max([limit, nsig * std]))
+    else:
+        LOGGER.debug("Outlier Filter loop exceeds maximum number of iterations."
+                     " Current filter-mask will be used.")
     return mask
 
 
 def _get_data_without_slope(mask, x, y):
+    """ Remove the slope on the data by performing a linear fit. """
     m, b = np.polyfit(x[mask], y[mask], 1)
     return y[mask] - b - m * x[mask], y - b - m * x
 
 
-def _get_data(mask, data):
-    return data[mask], data
-
-
-# Set the sigma cut, that expects 1 value to be cut
-# if it is sample of normal distribution
 def _get_significance_cut_from_length(length):
+    """ Set the sigma cut,
+    that expects 1 value to be cut if it is sample of normal distribution."""
```

Comment:
1? up for debate, but usually one writes number < 10 out, or in this replace with `a`?